File: Hoja8/src/Paciente.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Rodrigo Arriaza;
 */
public class Paciente implements Comparable<Paciente>{
    private String nombre, enfermedad, prioridad;

    public Paciente(String nombre, String enfermedad, String prioridad) {
        this.nombre = nombre;
        this.enfermedad = enfermedad;
        this.prioridad=prioridad;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getEnfermedad() {
        return enfermedad;
    }

    public void setEnfermedad(String enfermedad) {
        this.enfermedad = enfermedad;
    }

    public String getPriority() {
        return prioridad;
    }

    public void setPriority(String prioridad) {
        this.prioridad=prioridad;
    }

    @Override
    public int compareTo(Paciente otro) {
        return prioridad.compareTo(otro.getPriority());
    }

    public String toString(){
        return getNombre()+", "+getEnfermedad()+", "+getPriority();
    }
    
    
}

File: Hoja8/Paciente.scala
/**
  * Rodrigo Arriaza
  * Joice Miranda 
  */
class Paciente(nombre: String, enfermedad: String, prioridad: String) extends Comparable[Paciente]{
 // declaracion de variables con valores ingresados
  var name: String = nombre
  var _enfermedad: String = enfermedad
  var priority: String = prioridad

  //def nombre(): String = name
  //def enfermedad(): String = _enfermedad
  //def prioridad(): String = priority

  
     /**
     * @param no param
     * @return String nombre, String enfermedad, String priority
     * Devuelve el nombre, enfermedad y prioridad de un paciente
     */      
  override def toString(): String = {
    var cadena = ""
    cadena = nombre+ ", " + enfermedad + ", " + prioridad
    cadena
  }

  /**
     * @param Paciente paciente
     * @return int prioridad 
     * Devuelve el nivel de prioridad del paciente
     */
  override def compareTo(o: Paciente): Int = {
    priority.compareTo(o.priority)
    //o.prioridad().compareTo(priority)
  }
}
